banzai/utils/file_utils.py

```diff
@@ -85,4 +131,18 @@ def get_calibration_filename(image):
                 cal_file += '-{}'.format(filename_part)
         cal_file += '.fits'
         return cal_file
+
     return get_calibration_filename
+
+
+def download_from_s3(file_info, output_directory, runtime_context):
+    frame_id = file_info.get('frameid')
+    logger.info(f"Downloading file {file_info.get('filename')} from archive. ID: {frame_id}.",
+                extra_tags={'filename': file_info.get('filename')})
+    url = f'{runtime_context.ARCHIVE_FRAME_URL}/{frame_id}'
+
+    response = requests.get(url, headers=runtime_context.ARCHIVE_AUTH_TOKEN, stream=True).json()
```

Comment:
What is the stream=True here actually getting you? Parameter description is "if False, the response content will be immediately downloaded." If it's True then the download is delayed until the content is accessed, but in this case the content is being accessed right away https://requests.readthedocs.io/en/master/api/#main-interface

Reply:
You're absolutely right. Good catch.